test/foundry/Certify.t.sol

```diff
@@ -0,0 +1,253 @@
+
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Prank as Alice to grant role
+        vm.startPrank(alice);
+
+        // Grant CERTIFIER role to bob
+        vault.grantRole(vault.CERTIFIER(), bob);
+
+        // Prank as Bob for the transaction
+        vm.startPrank(bob);
+
+        // Expect the Certify event
+        vm.expectRevert(abi.encodeWithSelector(FutureReferenceBlock.selector, bob, fuzzedFutureBlockNumber));
+
+        // Call the certify function
+        vault.certify(certifyUntil, fuzzedFutureBlockNumber, forceUntil, data);
+
+        vm.stopPrank();
+    }
+
+    /// Test certify with force until true
+    function testCertifyWithForceUntilTrue(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 assets,
+        uint256 minShareRatio,
+        bytes memory data,
+        uint256 certifyUntil,
+        uint256 forceCertifyUntil,
+        uint256 referenceBlockNumber,
+        uint256 nextReferenceBlockNumber
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address alice = vm.addr((fuzzedKeyAlice % (SECP256K1_ORDER - 1)) + 1);
+
+        minShareRatio = bound(minShareRatio, 0, 1e18);
+        certifyUntil = bound(certifyUntil, 1, type(uint32).max);
+        forceCertifyUntil = bound(forceCertifyUntil, 1, type(uint32).max);
+        vm.assume(certifyUntil != forceCertifyUntil);
+
+        nextReferenceBlockNumber = bound(nextReferenceBlockNumber, 0, block.number);
+        referenceBlockNumber = bound(referenceBlockNumber, 0, block.number);
+        vm.assume(referenceBlockNumber != nextReferenceBlockNumber);
+
+        // Assume that assets are within a valid range
+        assets = bound(assets, 1, type(uint256).max);
+
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Prank as Alice to set role
+        vm.startPrank(alice);
+        vault.grantRole(vault.DEPOSITOR(), alice);
+        vault.grantRole(vault.CERTIFIER(), alice);
+
+        // Expect the Certify event
+        vm.expectEmit(false, false, false, true);
+        emit Certify(alice, certifyUntil, referenceBlockNumber, false, data);
+
+        // Certify system
+        vault.certify(certifyUntil, referenceBlockNumber, false, data);
+
+        // Expect the Certify event
+        vm.expectEmit(false, false, false, true);
+        emit Certify(alice, forceCertifyUntil, nextReferenceBlockNumber, true, data);
+
+        // Certify with forceUntil true
+        vault.certify(forceCertifyUntil, nextReferenceBlockNumber, true, data);
+
+        // Call deposit to make sure system is certified
+        vm.expectEmit(false, false, false, true);
+        emit DepositWithReceipt(alice, alice, assets, assets, 1, data);
+
+        // Attempt to deposit, should revert
```

Comment:
this says it should revert but i don't see an expect revert

for the forced certification we should have tests showing both successful and revert deposit, e.g. with a warp